packages/hash/api/src/graphql/resolvers/knowledge/entity/entity.ts

```diff
@@ -157,9 +158,26 @@ export const updatePersistedEntity: ResolverFn<
   { entityId, updatedProperties },
   { dataSources: { graphApi }, userModel },
 ) => {
-  const entityModel = await EntityModel.getLatest(graphApi, {
-    entityId,
-  });
+  // The user needs to be signed up if they aren't updating their own user entity
+  if (entityId !== userModel.entityId && !userModel.isAccountSignupComplete()) {
+    throw new ForbiddenError(
+      "You must complete the sign-up process to perform this action.",
+    );
+  }
+
+  const entityModel = await EntityModel.getLatest(graphApi, { entityId });
+
+  for (const beforeUpdateHook of beforeUpdateEntityHooks) {
+    if (
+      beforeUpdateHook.entityTypeId === entityModel.entityTypeModel.schema.$id
+    ) {
+      await beforeUpdateHook.callback({
```

Comment:
I am assuming if we want the fallibility here, we would have to act on the callback, and only conditionally continue

Reply:
I'm not sure I'm understanding, but if the callback fails the error is propagated into this scope, and therefore the function wouldn't continue. We don't have to enforce this manually in the current design.